apex/contrib/csrc/conv_bias_relu/conv_bias_relu.cpp

```diff
@@ -143,49 +143,53 @@ cudnn_frontend::ExecutionPlan& getOrCreatePlan(cudnnHandle_t handle_,
     DEBUG_CUDNN_MSG(log_buf, "Found plan in cache");
     return it->second;
   } else {
-    if (use_heuristic){
+    if (use_heuristic) {
       // TODO: confirm which mode to use
       auto heuristics = cudnn_frontend::EngineHeuristicsBuilder()
         .setOperationGraph(opGraph)
         .setHeurMode(CUDNN_HEUR_MODE_INSTANT)
         .build();
-      // try 3 times for now as WAR for no heuristic training
-      int max_tries = 3, count = 0;
-      auto& engine_configs = heuristics.getEngineConfig(max_tries);
-      while(true) {
+      auto engine_config_count = heuristics.getEngineConfigCount();
+      auto& engine_configs = heuristics.getEngineConfig(engine_config_count);
+      for (int64_t count = 0; count < engine_config_count; count++) {
         try {
           plan_cache.emplace(cache_string, std::move(cudnn_frontend::ExecutionPlanBuilder()
                                                      .setHandle(handle_)
                                                      .setEngineConfig(engine_configs[count], opGraph.getTag())
                                                      .build()));
           break;
         } catch (cudnn_frontend::cudnnException e) {
-          if (++count == max_tries) throw e;
+          // Throw exception if all engines failed
+          if (count == (engine_config_count - 1)) {
+            throw e;
+          } else {
+            continue;
+          }
         }
       }
-    }else{
-    DEBUG_CUDNN_MSG(log_buf, "No plan in cache");
-    // How many engines support this operation graph ?
-    auto total_engines = opGraph.getEngineCount();
-    DEBUG_CUDNN_MSG(log_buf, opGraph.describe() << " has " << total_engines << " engines.");
-    // We have to randomly pick one engine from [0, total_engines)
-    // Selecting "0" by default
-    auto engine = cudnn_frontend::EngineBuilder().setGlobalEngineIdx(0).setOperationGraph(opGraph).build();
-    DEBUG_CUDNN_MSG(log_buf, engine.describe());
-    auto& knobs = engine.getSupportedKnobs();
-    for (auto it = std::begin(knobs); it != std::end(knobs); ++it) {
-      DEBUG_CUDNN_MSG(log_buf, it->describe());
-    }
-    if (knobs.begin() != knobs.end()) {
-      DEBUG_CUDNN_MSG(log_buf, "Updated knob choice");
-      knobs.begin()->setChoice(knobs.begin()->getMinValue() + 1);
-      DEBUG_CUDNN_MSG(log_buf, knobs.begin()->describe());
-    }
+    } else {
+      DEBUG_CUDNN_MSG(log_buf, "No plan in cache");
```

Comment:
Nit: this is isn't from this PR, but I'm guessing that this should be moved outside this branch to just after the `return it->second` line, as both this case and the `use_heuristic` branch are for when there is no plan in the cache.

Reply:
Moved accordingly